src/ruby/yast/wfm.rb

```diff
@@ -273,7 +273,12 @@ def self.call_builtin_wrapper(*args)
         end
       else
         Yast.import "Report"
-        Report.Error(msg)
+        # Pure approximation here
+        # 50 is for usable text area width, +6 is for additional lines like
+        # button line, Error caption and so. Whole dialog is at most 20 lines
+        # high to fit into screen
+        height = [msg.size / 50 + 6, 20].min
```

Comment:
Hmm, how this will work with 80x24 terminal? Can you try it? Maybe you can query Display size also for reasonable maximum. But also please keep code KISS as it is last stand of defense when it goes bad.

Reply:
JFYI: `String#size` is not the same as width on the screen! That does not work esp in CJK languages: `ruby -e 'puts "日".size'` => 1, but actually the character has with 2 columns (in text mode it is wide as 2 characters).

But to make it simple let's keep it, the only disadvantage is that the user will have to scroll a bit more.